### PR TITLE
[API] m2a filtering fix

### PR DESCRIPTION
**Pull request by d1rOn**
Resolving an issue that causes MySQL to throw an error when trying to apply filter on a m2a collection.

### CODE REVIEW

File: api/src/utils/apply-query.ts [hunk 1]
```diff
@@ -221,7 +221,7 @@ export function applyFilter(
 							.on(
 								`${parentAlias || parentCollection}.${relation.field}`,
 								'=',
-								knex.raw(`CAST(?? AS VARCHAR(255))`, `${alias}.${schema.collections[pathScope].primary}`)
+								knex.raw(`CAST(?? AS CHAR)`, `${alias}.${schema.collections[pathScope].primary}`)
```
Comment (rijkvanzanten) [inline api/src/utils/apply-query.ts:224]: cc @Oreilles

Comment (Oreilles) [reply api/src/utils/apply-query.ts]: `CHAR` works for all vendors except Postgres, which returns only one character. Casting to `CHAR(255)` instead seems to make everyone happy.

**Review by oreilles — APPROVED** [review]

**Comment by rijkvanzanten** [comment]
@paescuj This PR changed an API file, but didn't kick in the E2E test. Any thoughts?

**Comment by paescuj** [comment]
> @paescuj This PR changed an API file, but didn't kick in the E2E test. Any thoughts?

E2E tests have been disabled for PRs in the past (I think because they weren't working properly). Now, that they do work I'm going to enable them again, okay?

Note: I'm working on a new feature for the `skip-duplicate-actions` which will allow us to skip or enforce individual jobs based on changed files (e.g. run E2E tests only if files in `/api` or `/tests` have been changed, ...)

**Comment by rijkvanzanten** [comment]
> E2E tests have been disabled for PRs in the past (I think because they weren't working properly). Now, that they do work I'm going to enable them again, okay?

Yes please! 🙏🏻

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**